src/Catalyst.Core.Lib/Util/Duration.cs

```diff
@@ -0,0 +1,175 @@
+                case "us":
+                case "µs":
+                    return TimeSpan.FromTicks((long) (value * TicksPerMicrosecond));
+                case "ns":
+                    return TimeSpan.FromTicks((long) (value * TicksPerNanosecond));
+                case "":
+                    throw new FormatException("Missing IPFS duration unit.");
+                default:
+                    throw new FormatException($"Unknown IPFS duration unit '{unit}'.");
+            }
+        }
+
+        static double ParseNumber(StringReader reader)
+        {
+            var s = new StringBuilder();
+            while (true)
+            {
+                var c = (char) reader.Peek();
+                if (char.IsDigit(c) || c == '.')
+                {
+                    s.Append(c);
+                    reader.Read();
+                    continue;
+                }
+
+                return double.Parse(s.ToString(), CultureInfo.InvariantCulture);
+            }
+        }
+
+        static string ParseUnit(StringReader reader)
+        {
+            var s = new StringBuilder();
+            while (true)
+            {
+                var c = (char) reader.Peek();
+                if (char.IsDigit(c) || c == '.' || c == (char) 0xFFFF)
+                    break;
+                s.Append(c);
+                reader.Read();
+            }
+
+            return s.ToString();
+        }
+
+        /// <summary>
+        ///   Converts the <see cref="TimeSpan"/> to the equivalent string representation of an 
+        ///   IPFS duration.
+        /// </summary>
+        /// <param name="duration">
+        ///   The <see cref="TimeSpan"/> to convert.
+        /// </param>
+        /// <param name="zeroValue">
+        ///   The string representation, when the <paramref name="duration"/> 
+        ///   is equal to <see cref="TimeSpan.Zero"/>/
+        /// </param>
+        /// <returns>
+        ///   The IPFS duration string representation.
+        /// </returns>
+        public static string Stringify(TimeSpan duration, string zeroValue = "0s")
+        {
+            if (duration.Ticks == 0)
+                return zeroValue;
+
+            var s = new StringBuilder();
+            if (duration.Ticks < 0)
+            {
+                s.Append('-');
+                duration = TimeSpan.FromTicks(-duration.Ticks);
+            }
+
+            Stringify(Math.Floor(duration.TotalHours), "h", s);
+            Stringify(duration.Minutes, "m", s);
+            Stringify(duration.Seconds, "s", s);
+            Stringify(duration.Milliseconds, "ms", s);
+            Stringify((long) ((double) duration.Ticks / TicksPerMicrosecond) % 1000, "us", s);
+            Stringify((long) ((double) duration.Ticks / TicksPerNanosecond) % 1000, "ns", s);
+
+            return s.ToString();
+        }
+
+        static void Stringify(double value, string unit, StringBuilder sb)
+        {
+            if (value == 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)